### PR TITLE
Career Page 

### PR DESCRIPTION
**Pull request by ChiragSinghal2001**
## Related Issue

Closes: #[issue number]

## Checklist:

- [ ] My code adheres to the established style guidelines of this project.
- [ ] I have conducted a self-review of my code.
- [ ] I have included comments in areas that may be difficult to understand.
- [ ] I have made corresponding updates to the project documentation.
- [ ] My changes have not introduced any new warnings.

## Screenshots

|           Original           |           Updated           |
| :--------------------------: | :-------------------------: |
| Original screenshot | Updated screenshot |

Please provide any necessary screenshots to illustrate the changes made.

### CODE REVIEW

File: components/JobApplicationForm.js [hunk 1]
```diff
@@ -0,0 +1,704 @@
...
+// import { useJobContext } from '../context/JobContext';
+// import { useRouter } from 'next/router';
+
+// function JobApplicationForm() {
+//   const router =useRouter();
+//   const { jobDetails } = useJobContext();
+//   const location = useLocation();
+//   const jobID = location.pathname.split('/').pop();
+//   const [captcha, setCaptcha] = useState('');
+//   const [captchaError, setCaptchaError] = useState('');
```
Comment (anmode) [inline components/JobApplicationForm.js]: remove these commented code

File: components/JobApplicationForm.js [hunk 2]
```diff
@@ -0,0 +1,704 @@
...
+// export default JobApplicationForm;
+
+import React, { useState, useEffect } from "react";
+import { useRouter } from "next/router";
+import { useJobContext } from "../context/JobContext";
+
+function JobApplicationForm() {
+  const router = useRouter();
+  const { jobDetails } = useJobContext();
+  const { jobID } = router.query; // Get jobID from the query parameters
```
Comment (anmode) [inline components/JobApplicationForm.js]: chatgpt :)

File: components/JobApplicationForm.js [hunk 3]
```diff
@@ -0,0 +1,704 @@
...
+
+  const generateCaptcha = () => {
+    const chars =
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+    let captcha = "";
+    for (let i = 0; i < 6; i++) {
+      captcha += chars.charAt(Math.floor(Math.random() * chars.length));
+    }
+    return captcha;
+  };
```
Comment (anmode) [inline components/JobApplicationForm.js]: why this captcha?

File: context/JobContext.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    {
+      jobid: 2,
+      country: "INDIA",
+      city: "BANGALORE",
+      title: "Full Stack Developer",
+      role: "Software Engineering - Project Role",
+      description: "Full Stack Development...",
+      businessArea: "Technology",
+      experience: "3-5",
+      posted: "Posted within last week",
```
Comment (anmode) [inline context/JobContext.js]: we have to remove this.. as we will fetching data from backend..

File: pages/career/applyjob/[jobID].page.js [hunk 5]
```diff
@@ -0,0 +1,468 @@
...
+    expectedSalary: "",
+    availableToJoin: "",
+    currentLocation: "",
+    notes: "",
+    rfa: "",
+    linkedin: "",
+    portfolio: "",
+    github: "",
+    consent: false,
+  });
```
Comment (anmode) [inline pages/career/applyjob/[jobID].page.js]: this form data is in job application form? and component almost code matched with this job id page code?

**Review by anmode — CHANGES_REQUESTED** [review]
Initial review... btw great job. thankss

File: components/JobApplicationForm.js (hunk 1 above)
Comment (ChiragSinghal2001) [reply components/JobApplicationForm.js]: Sure

File: components/JobApplicationForm.js (hunk 3 above)
Comment (ChiragSinghal2001) [reply components/JobApplicationForm.js]: For human verification

File: context/JobContext.js (hunk 4 above)
Comment (ChiragSinghal2001) [reply context/JobContext.js]: As of now, we're using hard coded data but soon it will be replaced

**Comment by anmode** [comment]
note: we have to create an API for updating, and creating job posts. We will have the UI in the admin dashboard.

**Comment by anmode** [comment]
One important to note: in future may be we will passing more URL params as refer id.. how we will handling those? Is it possible to handle that via using page routing.  can we simple have `index.js` in place of `[jobID].page.js`?

**Comment by ChiragSinghal2001** [comment]
> One important to note: in future may be we will passing more URL params as refer id.. how we will handling those? Is it possible to handle that via using page routing. can we simple have `index.js` in place of `[jobID].page.js`?

we can create something like ([...params].js) to catch all routes if dynamic params increase .

**Comment by ChiragSinghal2001** [comment]
> note: we have to create an API for updating, and creating job posts. We will have the UI in the admin dashboard.

The API to create job posts has already been added to the backend.

**Comment by anmode** [comment]
> > One important to note: in future may be we will passing more URL params as refer id.. how we will handling those? Is it possible to handle that via using page routing. can we simple have `index.js` in place of `[jobID].page.js`?
> 
> we can create something like ([...params].js) to catch all routes if dynamic params increase .

Umm.. this needs to be rethink. Let's connect to have discussion over this!

**Comment by ChiragSinghal2001** [comment]
> > > One important to note: in future may be we will passing more URL params as refer id.. how we will handling those? Is it possible to handle that via using page routing. can we simple have `index.js` in place of `[jobID].page.js`?
> > 
> > 
> > we can create something like ([...params].js) to catch all routes if dynamic params increase .
> 
> Umm.. this needs to be rethink. Let's connect to have discussion over this!

Sure

**Comment by ChiragSinghal2001** [comment]
Changes in UI in new commit

**Comment by anmode** [comment]
Some last small fix: 1. In case of no job do not show the header of the page ( which is open position...)( does not look ) 
2. 
Screenshot 2024-07-26 at 9 05 23 PM
black color looks very odd ( white would be good)

File: context/JobContext.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+const JobContext = createContext();
+
+export const JobProvider = ({ children }) => {
+  const [jobDetails, setJobDetails] = useState([]);
+  const [particularJob, setParticularJob] = useState(null);
+
+  useEffect(() => {
+    const fetchJobsData = async () => {
+      const response = await fetch(
+        `${process.env.NEXT_PUBLIC_BACKEND_URL}/api/jobspost/jobsVacancy`,
```
Comment (anmode) [inline context/JobContext.js]: Just a question why this routing? `api/job/all`?

File: pages/career/applyjob/[jobID].page.js [hunk 7]
```diff
@@ -0,0 +1,474 @@
+import React, { useState, useEffect } from "react";
+import { useRouter } from "next/router";
+import { useJobContext } from "../../../context/JobContext";
+import Button from "../../../components/UI/Button/Button";
```
Comment (anmode) [inline pages/career/applyjob/[jobID].page.js]: `pages/career/applyjob/[jobID].page.js` ---> `pages/career/apply/index.page.js`.  ( please can we have like this jobID=${jobId} we can query this param

File: pages/career/applyjob/[jobID].page.js [hunk 8]
```diff
@@ -0,0 +1,474 @@
...
+            headers: {
+              "Content-Type": "application/json",
+            },
+            body: JSON.stringify({
+              ...formData,
+              jobID: particularJob?.jobid,
+              jobTitle: particularJob?.title,
+            }),
+          },
+        );
```
Comment (anmode) [inline pages/career/applyjob/[jobID].page.js]: this is wrong.. please use axios post ..

File: pages/career/applyjob/[jobID].page.js [hunk 9]
```diff
@@ -0,0 +1,474 @@
...
+          toast.error(`Error submitting form: ${data.message}`);
+        }
+      } catch (error) {
+        toast.error("Network error, please try again later.");
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (Array.isArray(jobDetails) && jobID) {
```
Comment (anmode) [inline pages/career/applyjob/[jobID].page.js]: what this means ? `Array.isArray`? can we check that jobDetails are not null?

File: pages/career/applyjob/[jobID].page.js [hunk 10]
```diff
@@ -0,0 +1,474 @@
...
+              type="text"
+              id="currentLocation"
+              name="currentLocation"
+              value={formData.currentLocation}
+              onChange={handleChange}
+              className="tw-font-sans tw-w-full tw-px-3 tw-py-2 tw-border tw-border-gray-300 tw-rounded-lg tw-bg-gray-100 tw-text-gray-800"
+              placeholder="Aligarh"
+            />
+          </div>
+
```
Comment (anmode) [inline pages/career/applyjob/[jobID].page.js]: we have library for dropdown showing all locations. So user can easily select it's location

File: pages/career.page.js [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+      <Section
+        kicker="Join Our Team"
+        heading="Open Positions"
+        subheading="Discover a Place Where Your Passion and Skills Drive Success"
+        align="center"
+        className="tw-mt-10 tw-pt-12 tw-pb-0 tw-mb-0"
+      />
+      <main className="tw-flex tw-flex-col tw-items-center tw-justify-center tw-px-4 tw-bg-[#f1f5f9]">
+        <div className=" tw-font-sans tw-w-full tw-max-w-6xl tw-mx-auto tw-text-center">
+          {Array.isArray(jobDetails) && jobDetails.length > 0 ? (
```
Comment (anmode) [inline pages/career.page.js]: in case of jobDetails is array of null items.. will it work?

**Review by anmode — CHANGES_REQUESTED** [review]
Some last comment.. Nd is it mobile responsive?

**Comment by ChiragSinghal2001** [comment]
> Some last small fix: 1. In case of no job do not show the header of the page ( which is open position...)( does not look ) 2. Screenshot 2024-07-26 at 9 05 23 PM black color looks very odd ( white would be good)

okk sure

**Comment by ChiragSinghal2001** [comment]
> Some last comment.. Nd is it mobile responsive?

yes it is

**Comment by ChiragSinghal2001** [comment]
UI is improved and a new page added

File: pages/career.page.js (hunk 11 above)
Comment (ChiragSinghal2001) [reply pages/career.page.js]: I changed it for better readability

**Review by anmode — APPROVED** [review]

**Comment by anmode** [comment]
@ChiragSinghal2001 cool, Really amazing thanks for making this 🚀 . Please make sure to take it down your API key as this repo is open-source. We will create our own API key.

**Pull request merged**